src/Servers/Kestrel/Kestrel/test/ConfigurationReaderTests.cs

```diff
@@ -68,6 +68,39 @@ public void ReadCertificatesSection_ReturnsCollection()
             Assert.True(storeCert.AllowInvalid);
         }
 
+        [Fact]
+        public void ReadCertificatesSection_IsCaseInsensitive()
+        {
+            var config = new ConfigurationBuilder().AddInMemoryCollection(new[]
+            {
+                new KeyValuePair<string, string>("Certificates:filecert:Path", "/path/cert.pfx"),
+                new KeyValuePair<string, string>("Certificates:FILECERT:Password", "certpassword"),
```

Comment:
This should still pass if the entire key is UPPERCASE, right? I know this PR only changes the dictionary comparer, but I see no reason not to verify the rest of it case insensitive while we're at it.